saleor/product/utils/variant_prices.py

```diff
@@ -0,0 +1,47 @@
+from prices.taxed_money import TaxedMoney
+
+from ...discount.utils import fetch_active_discounts
+from ..models import Product
+
+
+def _get_product_minimal_variant_price(product, discounts):
+    # Start with the product's price as the minimal one
+    minimal_variant_price = product.price
+    for variant in product.variants.all():
+        variant_price = variant.get_price(discounts=discounts)
```

Comment:
`get_price` will return normal `Money`. No need to cast it. All `taxes` logic has been extracted from `get_price`